modules/promotions/assets/js/react/behavior.js

```diff
@@ -0,0 +1,87 @@
+import App from './app';
+import { createRoot } from 'react-dom/client';
+
+export default class ReactPromotionBehavior extends Marionette.Behavior {
+	promotionInfoTip = null;
+
+	selectors = {
+		reactAnchor: '.e-promotion-react',
+	};
+
+	ui() {
+		return {
+			animatedHeadlineButton: '[data-promotion].elementor-control-type-switcher',
+		};
+	}
+
+	events() {
+		return {
+			'click @ui.animatedHeadlineButton': 'onClickControlButtonAnimatedHeadline',
+		};
+	}
+
+	promotionData( promotionType ) {
+		return elementorPromotionsData[ promotionType ] || {};
+	}
+
+	onClickControlButtonAnimatedHeadline( event ) {
+		event.stopPropagation();
+		this.mount();
+	}
+
+	mount() {
+		if ( this.promotionInfoTip ) {
+			return;
+		}
+
+		const rootElement = document.querySelector( this.selectors.reactAnchor );
+
+		if ( ! rootElement ) {
+			return;
+		}
+
+		this.attachEditorEventListeners();
+
+		this.promotionInfoTip = createRoot( rootElement );
+
+		const colorScheme = elementor?.getPreferences?.( 'ui_theme' ) || 'auto',
+			isRTL = elementorCommon.config.isRTL,
+			promotionType = rootElement.getAttribute( 'data-promotion' );
+
+		this.promotionInfoTip.render(
+			<App
+				colorScheme={ colorScheme }
+				isRTL={ isRTL }
+				promotionsData={ this.promotionData( promotionType ) }
+				onClose={ () => this.unmount() }
+			/>,
+		);
+	}
+
+	unmount() {
+		if ( this.promotionInfoTip ) {
+			this.detachEditorEventListeners();
+			this.promotionInfoTip.unmount();
+		}
+
+		this.promotionInfoTip = null;
+	}
+
+	onRoute = () => {};
```

Comment:
Do we need this?

Reply:
@louiswol94 this way the event was set/removed properly.. not sure why. making the `onRoute` worked for me, and `$e....on(...)` + `$e...off(...)` worked fine...

Reply:
`$e.routes.on( 'run:after', this.onRoute.bind( this ) );` - maybe write it like this in `attachEditorEventListeners`? And write the function logic in here ☝️

Reply:
for some reason other combinations didn't result in a success during the `off` call, and even handler was not removed and continued to listen to the events.